### ISSUE TITLE
simexp crashes if first truth file is missing a target class

### ISSUE THREAD

**Issue by sbailey** [desihub/desisim#489]
As originally discovered by @julianbautista (full traceback below):

`desispec.simexp.get_mock_spectra` crashes if the first truth file is missing a target class that a later truth file does have.  In Julian's case, he's processing tile 12725 which covered nside=64 healpixels [4989, 4990, 4991, 5076...].  `/global/cscratch1/sd/jbautist/desi/dev/end2end_lya/targets/49/4989/truth-64-4989.fits` does not have a TRUTH_WD HDU, while the next pixel `/global/cscratch1/sd/jbautist/desi/dev/end2end_lya/targets/49/4990/truth-64-4990.fits` does.

[simexp.py lines 773-774]:(https://github.com/desihub/desisim/blob/a7daab77a60a6a8abab386c68fee36450e3eb7b8/py/desisim/simexp.py#L773-L774) populate the `objmeta` dictionary with the keys found in the first file:
```python
            for key in tmpobjmeta.keys():
                objmeta[key] = list()
```
But then when it gets to a new file with a new key it crashes on lines [781-782](https://github.com/desihub/desisim/blob/a7daab77a60a6a8abab386c68fee36450e3eb7b8/py/desisim/simexp.py#L781-L782) appending values:
```python
        for key in tmpobjmeta.keys():
            objmeta[key].append(tmpobjmeta[key])
```

@julianbautista I think the fix is just to check they keys from every file, e.g. something like
```python
        for key in tmpobjmeta.keys():
            if key not in objmeta:
                objmeta[key] = list()

            objmeta[key].append(tmpobjmeta[key])
```
@julianbautista could you give that a try and test it with your case?

<hr/>
Original report with traceback:
```
FAILED: night 20200323 expid 317; see /global/cscratch1/sd/jbautist/desi/dev/end2end_lya/spectro/sim/mini_lya/20200323/00000317/simspec-00000317.log

Looking into the file above I find: 

INFO:newexp_mock.py:36:parse: /global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desisim/0.31.2/bin/newexp-mock --obslist /global/cscratch1/sd/jbautist/desi/dev/end2e
nd_lya/survey/exposures.fits --fiberassign /global/cscratch1/sd/jbautist/desi/dev/end2end_lya/fiberassign --mockdir /global/cscratch1/sd/jbautist/desi/dev/end2end_lya/targets --nside 6
4 --clobber --expid 317
INFO:newexp_mock.py:80:main: Simulating night 20200323 expid 317 tile 12725
CRITICAL:newexp_mock.py:85:main: Failed expid 317 fiberassign /global/cscratch1/sd/jbautist/desi/dev/end2end_lya/fiberassign/tile-12725.fits tile 12725
Traceback (most recent call last):
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desisim/0.31.2/bin/newexp-mock", line 4, in <module>
    __import__('pkg_resources').run_script('desisim==0.31.2', 'newexp-mock')
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/conda/lib/python3.6/site-packages/pkg_resources/__init__.py", line 654, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/conda/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1434, in run_script
    exec(code, namespace, namespace)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desisim/0.31.2/lib/python3.6/site-packages/desisim-0.31.2-py3.6.egg/EGG-INFO/scripts/newexp-mock", line 9, 
in <module>
    sys.exit(main())
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desisim/0.31.2/lib/python3.6/site-packages/desisim-0.31.2-py3.6.egg/desisim/scripts/newexp_mock.py", line 8
6, in main
    raise err
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desisim/0.31.2/lib/python3.6/site-packages/desisim-0.31.2-py3.6.egg/desisim/scripts/newexp_mock.py", line 8
2, in main
    flux, wave, meta, objmeta = get_mock_spectra(fiberassign, mockdir=args.mockdir, nside=args.nside)
  File "/global/common/software/desi/cori/desiconda/20180709-1.2.6-spec/code/desisim/0.31.2/lib/python3.6/site-packages/desisim-0.31.2-py3.6.egg/desisim/simexp.py", line 782, in get_mo
ck_spectra
    objmeta[key].append(tmpobjmeta[key])
KeyError: 'WD'
```

**Comment by julianbautista** [comment]
I tested this change and all works. I submitted a PR with this fix. Thanks!

**Comment by sbailey** [comment]
Fixed by PR #490.  Closing.
